gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -117,13 +121,22 @@ public Task(TaskContext context, TaskStateTracker taskStateTracker, TaskExecutor
         closer.register(new InstrumentedExtractorDecorator<>(this.taskState, this.taskContext.getExtractor()));
 
     this.converter = closer.register(new MultiConverter(this.taskContext.getConverters()));
-
+    try {
+      this.rowChecker = closer.register(this.taskContext.getRowLevelPolicyChecker());
```

Comment:
Why is this necessary? Shouldn't the converter instantiation be also in the try-catch block?